src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/CompletionHandlerTest.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer;
+using Microsoft.VisualStudio.LanguageServer.Client;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using Xunit;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    public class CompletionHandlerTest
+    {
+        public CompletionHandlerTest()
+        {
+            var joinableTaskContext = new JoinableTaskContextNode(new JoinableTaskContext());
+            JoinableTaskContext = joinableTaskContext.Context;
+            Uri = new Uri("C:/path/to/file.razor");
+            LSPDocumentSynchronizer = Mock.Of<LSPDocumentSynchronizer>(s => s.TrySynchronizeVirtualDocumentAsync(It.IsAny<LSPDocumentSnapshot>(), It.IsAny<VirtualDocumentSnapshot>(), It.IsAny<CancellationToken>()) == Task.FromResult(true));
+            LanguageClientBroker = Mock.Of<ILanguageClientBroker>();
+        }
+
+        public JoinableTaskContext JoinableTaskContext { get; }
+
+        private Uri Uri { get; }
+
+        private LSPDocumentSynchronizer LSPDocumentSynchronizer { get; }
+
+        private ILanguageClientBroker LanguageClientBroker { get; }
+
```

Comment:
Should add tests validating that C#/HTML projections with incompatible trigger characters results in `null`